manifests/virtual_machine.pp

```diff
@@ -132,11 +137,14 @@
           --extra-args 'auto netcfg/disable_dhcp=true'
         | VIRT_INSTALL_EOF
     }
+  }
 
-    exec { "${prefix}::autostart":
-      require => Exec["${prefix}::virt-install"],
-      creates => "${autostart_path}/${title}.xml",
-      command => "/usr/bin/virsh autostart ${title}",
-    }
+  exec { "/usr/bin/virsh set-lifecycle-action ${title} crash restart --config":
+    onlyif => "/usr/bin/virsh dumpxml --inactive ${title} | grep on_crash | grep -q destroy",
+  }
+
+  exec { "/usr/bin/virsh set-lifecycle-action ${title} crash restart --live":
+    onlyif  => "/usr/bin/virsh dumpxml ${title} | grep on_crash | grep -q destroy",
+    require => Exec["/usr/bin/virsh set-lifecycle-action ${title} crash restart --config"],
   }
```

Comment:
1. Not sure what exactly this does, but from the `--inactive` in one and not the other it looks like these are the both do the same thing but one is for stopped guests and the other is for running guests?
2. Shouldn't these be in the `if` above, so they don't run before or during vm creation?
3. What's the purpose of the require? Unless I was mistaken in (1) above, the order of these can't matter because they are mutually exclusive.

Reply:
1. I'll add a comment that hopefully clarifies, but the first one is "if the config xml file says `<on_crash>destroy</on_crash>`, change the file to `restart`." The second one behaves differently depending on whether the VM is running. If it's not running, `dumpxml` and `dumpxml --inactive` do the same thing, so the second exec won't run. If the VM is running, this checks the live xml and changes that if necessary.
2. In this case it doesn't matter, because the `dumpxml` command will fail if the VM doesn't exist yet, preventing the `exec` from running. The autostart bit is different because it runs when a symlink hasn't been created, which will lead to it executing before the VM is created if it's neither requiring the vm creation nor in a different condition branch.
3. The require is probably not required, but if it weren't there, it could result in `set-lifecycle-action --live` getting run on a VM that isn't running, and I'm not sure what that would do. Could result in a failing puppet resource alert? Didn't want to find out.